graphbolt/include/graphbolt/serialize.h

```diff
@@ -66,6 +66,40 @@ namespace graphbolt {
 torch::IValue read_from_archive(
     torch::serialize::InputArchive& archive, const std::string& key);
 
+/**
+ * @brief Read dictionary from archive.
+ * @param archive Input archive.
+ * @param key Key name of dictionary.
+ *
+ * @return dictionary.
+ */
+template <typename DictT>
```

Comment:
The PR makes a lot of sense to me. A minor suggestion is to make both key and value types as the templates for this function.
```cpp
template <typename KeyType, typename ValueType>
```
Then the function is general to all dictionary types and we don't need to specially handle `int64_t` and `torch::Tensor`.

Reply:
I wish I could achieve. The major blocker is the caller function name when converting IValue to real data such as `toInt()`, `toTensor()`. Is there any template-like calls for these ones? For example, `to<int>`, `to<Tensor>`?

Reply:
I see. I am not sure whether [this](https://github.com/pytorch/pytorch/blob/4ea7430ffb0b37e84108c70f90f402f5fd37262e/aten/src/ATen/core/ivalue.h#L1004) is workable.

Reply:
this works. let me further refine related code.

Reply:
I used generalized `to<>` for further de-duplication